apps/api/src/app/workflows-v2/usecases/placeholder-enrichment/payload-preview-value-generator-usecase.ts

```diff
@@ -0,0 +1,60 @@
+import { GeneratePreviewRequestDto, TipTapNode } from '@novu/shared';
+import { TransformPlaceholderMapUseCase } from './transform-placeholder-usecase';
+import {
+  CollectPlaceholdersFromTipTapSchemaUsecase,
+  extractPlaceholders,
+} from './collect-placeholders-from-tip-tap-schema.usecase';
+
+// Command interface with the same name as the class but without the "UseCase" suffix
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export interface AddKeysToPayloadBasedOnHydrationStrategyCommand {
+  dto: GeneratePreviewRequestDto;
+  controlValueKey: string;
+}
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** As previously stated, this should extend from `BaseCommand` to get validation issue parsing via `class-validators`‏